### ISSUE TITLE
Labels not added to deployment when in okteto.yaml manifest

### ISSUE THREAD

**Issue by nfrush** [okteto/okteto#1770]
**Describe the bug**
When using okteto to spin up a development container (i.e. okteto up) including a labels declaration within my okteto.yml manifest results in the following error:

deployment for labels 'opa_istio=enabled' not found

**To Reproduce**

1. Create a new okteto.yaml manifest
2. Add a labels block like the following:
```
labels:
  opa_istio: "enabled"
```
3. Run okteto up

Full okteto.yaml
```
name: okt-test
labels:
  opa_istio: "enabled"
autocreate: true
image: ANY_OKTETO_IMAGE
command: /bin/bash
sync:
- .:/usr/src/app
- customize:/customize
forward:
- 9239:9229
- 8088:8080
```

**Expected behavior**

I expected labels to be added to the Deployment that okteto ends up creating

**Desktop (please complete the following information):**

- OS: [e.g. macOS] Amazon Linux
- Version [e.g. 22] 2

**Additional context**
Add any other context about the problem here (e.g. okteto.yml, k8s manifests, logs)

**Comment by pchico83** [comment]
@nfrush `labels`, as they work today, are to select the target deployment you want to put on development mode:
https://okteto.com/docs/reference/manifest/#labels-mapstringstring-optional

We should probably rename it to `selector`, and let `labels` to define dev labels. But it is a breaking change...

@jLopezbarb one option is to implement `selector`, and show a warning when using `labels`

**Comment by freeo** [comment]
okteto 1.14.8 here.

As per the new warning message (labels being deprecated) I've added
```
selector:
  mykey: myvalue
```
and expected this to show up as label in the deployment+pod, so that I can set the label explicitly for my service manifests. But those don't show up anywhere, just the default labels generated by okteto, e.g. 
```
app=appname
interactive.dev.okteto.com=appname
pod-template-hash=deadbeef
```

Am I missing something here?

**Comment by pchico83** [comment]
@freeo yes, it's expected. `selector` is to select your deployment. If you want to add labels, add:
```
   metadata:
       labels:
          mykey: myvalue
```
to your okteto manifest.
https://okteto.com/docs/reference/manifest/#metadata-object-optional
